src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -143,6 +143,7 @@ protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersP
             var provenHeadersPayload = new ProvenHeadersPayload();
 
             ChainedHeader header = this.GetLastHeaderToSend(fork, getHeadersPayload.HashStop);
+            this.logger.LogDebug("Last header that will be sent in headers payload is '{0}'.", header);
```

Comment:
Do we want to keep all those debug logs?

Reply:
Yeah. 
I'm trying to separate logs to trace and debug because trace logs are too verbose but we have nothing on debug level.

Reply:
We probably need to also do that in many other places int eh code
We have no formal way to decide when we use debug logs